tools/Vitis-AI-Quantizer/vai_q_pytorch/csrc/cuda/nndct_cuda_math.cu

```diff
@@ -227,16 +236,20 @@ static void _vec_transform_reduce_inplace(const int dim,Dtype* data,
     __syncthreads();
   }
 
-  // Reduce last warp. Threads implicitly synchronized within a warp.
+  // Reduce last warp.
   if (tid < warpSize) {
     for (int shift = warpSize; shift > 0; shift >>= 1) {
       sdata[tid] = op.Reduce(sdata[tid], sdata[tid + shift]);

```

Comment:
@elliott-xilinx, this code here is reading and writing to shared memory from each thread. So, you need a `__threadfence_block()` to flush the shared memory writes **even on the CUDA side**.